nemo_skills/pipeline/utils.py

```diff
@@ -105,8 +105,25 @@ def get_generation_command(server_address, generation_commands):
     return cmd
 
 
+nemo_server_start_template = (
```

Comment:
Let's also create a new get_reward_server_command or some other name, which would be an equivalent of get_server_command but for rm. Or add a new parameter to get_server_command that will switch between llm/rm servers - this option is better if most of the parameters are expected to be the same (which is probably true, so let's maybe start with this)

Reply:
Here, I prefer the first option because it is more composable. The second option will result in gnarly nested if/else's of the form:

```
if server_type == 'nemo':
  if generate_type == 'generate':
    do_nemo_generate_thing()
  elif generate_type == 'rm':
    do_nemo_rm_thing()
if server_type == 'trtllm':
  if generate_type == 'generate':
    do_trtllm_generate_thing()
  elif generate_type == 'rm':
    raise ValueError("rm not supported for trtllm")
```

And it just gets really tricky to parse and update.

Reply:
Whereas with the first option would look more like:

```
def get_reward_server_command(server_type):
  if server_type == 'nemo':
    do_nemo_rm_thing()
  else:
    raise ValueError(f'unsupported reward model server_type={server_type}')

def get_server_command(server_type):
  if server_type == 'nemo':
    do nemo_generate_thing()
  elif server_type == 'trtllm':
    do_trtllm_generate_thing()
  elif server_type == 'vllm':
    do_vllm_generate_thing()
  else:
    raise ValueError(f'unsupported generation server_type={server_type}')
 
server_factories = {
  'generate': get_server_command,
  'rm': get_reward_server_command,
} 

def pipeline(server_type, generate_type):
  server_factory = server_factories.get(generate_type)
  if server_factory is None:
    raise ValueError(f'unsupported generate_type={generate_type}')
  server = server_factory(server_type)
```